.github/workflows/ostests-nightly.yaml

```diff
@@ -0,0 +1,95 @@
+# Nightly trigger for OS testing. See ../../hack/ostests/README.md for details.
+
+name: "OS tests :: Nightly"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 30 2 * * *
+
+env:
+  DISTRIBUTIONS: >-
+    [
+      ["alpine_3_17"],
+      ["centos_7", "centos_8", "centos_9"],
+      ["debian_10", "debian_11", "debian_12"],
+      ["fcos_38"],
+      ["fedora_38"],
+      ["flatcar"],
+      ["oracle_7_9", "oracle_8_7", "oracle_9_1"],
+      ["rhel_7", "rhel_8", "rhel_9"],
+      ["rocky_8", "rocky_9"],
+      ["ubuntu_2004", "ubuntu_2204", "ubuntu_2304"]
+    ]
+  NETWORK_PROVIDERS: >-
+    [
+      "kuberouter",
+      "calico"
+    ]
+
+jobs:
+  select:
+    name: Select
+    runs-on: ubuntu-latest
+
+    outputs:
+      os: ${{ steps.select.outputs.os }}
+      network-provider: ${{ steps.select.outputs.network-provider }}
+
+    steps:
+      # Generate a "well-distributed" list of all possible parameter
+      # combinations and select an entry based on the current day. This approach
+      # ensures that, if run once each day, there is a significant variation in
+      # parameter values each day while also guaranteeing that every combination
+      # will be chosen at some point.
```

Comment:
Did I understand this correctly: We have 22 OSes and 2 network providers on the matrix, this means 44 items. So each combo, say Alpine 3.17 with Calico, will be tested every 44 days?

Reply:
<details>
<summary>Starting from today, this would be list of combinations for each day:</summary>

```text
rocky_9/kuberouter
ubuntu_2204/calico
centos_9/kuberouter
debian_12/calico
oracle_9_1/kuberouter
rhel_9/calico
ubuntu_2304/kuberouter
alpine_3_17/calico
centos_7/kuberouter
debian_10/calico
fcos_38/kuberouter
fedora_38/calico
flatcar/kuberouter
oracle_7_9/calico
rhel_7/kuberouter
rocky_8/calico
ubuntu_2004/kuberouter
centos_8/calico
debian_11/kuberouter
oracle_8_7/calico
rhel_8/kuberouter
rocky_9/calico
ubuntu_2204/kuberouter
centos_9/calico
debian_12/kuberouter
oracle_9_1/calico
rhel_9/kuberouter
ubuntu_2304/calico
alpine_3_17/kuberouter
centos_7/calico
debian_10/kuberouter
fcos_38/calico
fedora_38/kuberouter
flatcar/calico
oracle_7_9/kuberouter
rhel_7/calico
rocky_8/kuberouter
ubuntu_2004/calico
centos_8/kuberouter
debian_11/calico
oracle_8_7/kuberouter
rhel_8/calico
```
</details>

Reply:
Specific combinations or bulk tests for many configurations may be triggered manually via the matrix job, if needed.